src/MongoDB.EntityFrameworkCore/Extensions/MongoIndexBuilderExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using MongoDB.EntityFrameworkCore.Metadata;
+
+namespace MongoDB.EntityFrameworkCore.Extensions;
+
+/// <summary>
+/// MongoDB specific extension methods for <see cref="IndexBuilder" />.
+/// </summary>
+public static class MongoIndexBuilderExtensions
+{
+    /// <summary>
+    /// Configures the <see cref="CreateIndexOptions"/> for the index.
+    /// </summary>
+    /// <param name="indexBuilder">The builder for the index being configured.</param>
+    /// <param name="options">The <see cref="CreateIndexOptions"/> for this index.</param>
+    /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+    /// <returns>
+    /// The same builder instance if the configuration was applied, <see langword="null" /> otherwise.
```

Comment:
Doesn't this return the input builder nevertheless and not null? Same question for the other methods

Reply:
It should, but this is the method signature and docs that EF's own builders use everywhere so I did not want to deviate from that.

Reply:
In this case shouldn't we follow the conventions and return null if the configuration is not applied?

Reply:
I'm just following the same patterns for extension methods used across EF - e.g. https://github.com/dotnet/efcore/blob/main/src/EFCore.Cosmos/Extensions/CosmosModelBuilderExtensions.cs#L55 which is the same sort of thing for Cosmos.

Reply:
Oops, you're totally right, now fixed.